dev/src/types.ts

```diff
@@ -27,9 +30,62 @@ export interface ApiMapValue {
   [k: string]: google.firestore.v1.IValue;
 }
 
-// We don't have type information for the JavaScript GapicClient.
-// tslint:disable-next-line:no-any
-export type GapicClient = any;
+/**
+ * The subset of methods we use from FirestoreClient.
+ *
+ * We don't depend on the actual Gapic client to avoid loading the GAX stack at
+ * module initialization time.
+ */
+export interface GapicClient {
+  getProjectId(): Promise<string>;
+  beginTransaction(
+    request: api.IBeginTransactionRequest,
+    options?: CallOptions
+  ): Promise<[api.IBeginTransactionResponse, unknown, unknown]>;
+  commit(
+    request: api.ICommitRequest,
+    options?: CallOptions
+  ): Promise<[api.ICommitResponse, unknown, unknown]>;
+  rollback(
+    request: api.IRollbackRequest,
+    options?: CallOptions
+  ): Promise<[google.protobuf.IEmpty, unknown, unknown]>;
+  batchGetDocuments(
+    request?: api.IBatchGetDocumentsRequest,
+    options?: CallOptions
+  ): Duplex;
```

Comment:
I am just returning a Duplex stream since I my unit tests do not create a Cancellable stream.